src/webAppZipPublisher.ts

```diff
@@ -0,0 +1,307 @@
+        this._folderPath = folderPath;
+    }
+
+    async prompt(): Promise<void> {
+        if (this._zipFilePath) {
+            this._zipCreatedByStep = false;
+            return;
+        }
+
+        if (!this._folderPath) {
+            if (!vscode.workspace.workspaceFolders) {
+                throw new Error('No open folder.');
+            }
+
+            const folderQuickPickItems = vscode.workspace.workspaceFolders.map((value) => {
+                {
+                    return <QuickPickItemWithData<vscode.WorkspaceFolder>>{
+                        label: value.name,
+                        description: '',
+                        data: value
+                    }
+                }
+            });
+            const folderQuickPickOption = { placeHolder: `Select the folder to Zip and deploy. (${this.stepProgressText})` };
+            const pickedItem = folderQuickPickItems.length == 1 ? 
+                folderQuickPickItems[0] : await this.showQuickPick(folderQuickPickItems, folderQuickPickOption);
+            this._folderPath = pickedItem.data.uri.fsPath;
+        }
+
+        let deployNumber = Math.floor(os.uptime());
+        while (true) {
+            this._zipFilePath = path.join(os.tmpdir(), `vscdeploy${deployNumber}.zip`);
+            const fileExists = await new Promise<boolean>((resolve, reject) => fs.exists(this._zipFilePath, exists => resolve(exists)));
+            if (!fileExists) {
+                break;
+            }
+            deployNumber++;
+        }
+    }
+
+    async execute(): Promise<void> {
+        if (this._zipCreatedByStep) {
+            this.wizard.writeline(`Creating Zip file for deployment: ${this.zipFilePath} ...`)
+            await this.zipDirectory(this.zipFilePath, this._folderPath, path.sep);
+        }
+    }
+
+    get zipFilePath(): string {
+        return this._zipFilePath;
+    }
+
+    get zipCreatedByStep(): boolean {
+        return this._zipCreatedByStep;
+    }
+
+    private zipDirectory(zipFilePath: string, sourcePath: string, targetPath: string): Promise<void> {
+        if (!sourcePath.endsWith(path.sep)) {
+            sourcePath += path.sep;
+        }
+        return new Promise((resolve, reject) =>{
+            const zipOutput = fs.createWriteStream(zipFilePath)
+            zipOutput.on('close', () => resolve());
+
+            const zipper = archiver('zip', { zlib: { level: 9 }});
+            zipper.on('error', err => reject(err));
+            zipper.pipe(zipOutput);
+            zipper.directory(this._folderPath, path.sep);
+            zipper.finalize();
+        });
+    }
+}
+
+class SubscriptionStep extends SubscriptionStepBase {
+    constructor(wizard: WizardBase,
+        azureAccount: AzureAccountWrapper,
+        subscription?: SubscriptionModels.Subscription) {
+        super(wizard, 'Select target Subscription', azureAccount, subscription);
+    }
+
+    async prompt(): Promise<void> {
+        if (!!this.subscription) {
```

Comment:
If there's no subscription, should it prompt the user to login?

Reply:
This line is to check if a subscription has been pre-set (for example, if the wizard was launched from the explorer, which has the context of subscription), if there is a pre-set subscription, then skip the prompt.